drivers/hwinfo/hwinfo_rpi_pico.c

```diff
@@ -29,6 +31,20 @@
 ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
 {
 	uint8_t id[FLASH_RUID_DATA_BYTES];
+
+#if CONFIG_HWINFO_RPI_PICO_CHIP_ID
+	rom_get_sys_info_fn get_sys_info = (rom_get_sys_info_fn)
+					rom_func_lookup_inline(ROM_FUNC_GET_SYS_INFO);
+	/* Words: CHIP_INFO, PACKAGE_SEL, DEVICE_ID, WAFER_ID */
+	uint32_t words[4];
+
+	int n = get_sys_info(words, ARRAY_SIZE(words), SYS_INFO_CHIP_INFO);
```

Comment:
This assertion doesn't make a lot of sense, but the variable dangles if asserts are disabled, should I remove this?

Reply:
Asserts are used if it could fail due to a bug or something.
If it cannot fail due to some runtime error, an assert is perfectly fine.